packages/renderer/src/lib/ImageRow.svelte

```diff
@@ -0,0 +1,101 @@
+<script lang="ts">
+import { router } from 'tinro';
+import ImageActions from './image/ImageActions.svelte';
+import type { ImageInfoUI } from './image/ImageInfoUI';
+import PushImageModal from './image/PushImageModal.svelte';
+import RenameImageModal from './image/RenameImageModal.svelte';
+import ImageIcon from './images/ImageIcon.svelte';
+import StatusIcon from './images/StatusIcon.svelte';
+
+export let object: any;
+export let column: number;
+
+$: image = object;
+
+export let multipleEngines = false; // TODO
+
+function openDetailsImage(image: ImageInfoUI) {
+  router.goto(`/images/${image.id}/${image.engineId}/${image.base64RepoTag}/summary`);
+}
+
+let pushImageModal = false;
+let pushImageModalImageInfo: ImageInfoUI | undefined = undefined;
+function handlePushImageModal(imageInfo: ImageInfoUI) {
+  pushImageModalImageInfo = imageInfo;
+  pushImageModal = true;
+}
+
+let renameImageModal = false;
+let renameImageModalImageInfo: ImageInfoUI | undefined = undefined;
+function handleRenameImageModal(imageInfo: ImageInfoUI) {
+  renameImageModalImageInfo = imageInfo;
+  renameImageModal = true;
+}
+
+function closeModals() {
+  pushImageModal = false;
+  renameImageModal = false;
+}
+</script>
+
+{#if column === 0}
```

Comment:
I don't like much this check. If we add a new column between 0 and 1, we need to edit all *Row components.
Wouldn't it be better to use `slots`?

Reply:
Good point. I thought of using slots earlier, the problem is that slots are fixed and named - i.e. we need to pick how many and what they're called up front. I'm trying other options, so far I think the best is just using names instead of numbers here.

Reply:
Pushed another update, columns are named now (via the column header) instead of numbered. This is makes the code a little more clearer and adding/removing columns in the future is straightforward.